pkg/resources/kafka/kafka.go

```diff
@@ -438,41 +440,132 @@ func arePodsAlreadyDeleted(pods []corev1.Pod, log logr.Logger) bool {
+		len(serverSecret.Data[v1alpha1.PasswordKey]) == 0 {
+		return false
+	}
+	return true
+}
+
+func getListenerSSLCertSecret(client client.Reader, commonSpec v1beta1.CommonListenerSpec, clusterName string, clusterNamespace string) (*corev1.Secret, error) {
+	// Use default SSL cert secret
+	secretNamespacedName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerServerCertTemplate, clusterName), Namespace: clusterNamespace}
+	if commonSpec.GetServerSSLCertSecretName() != "" {
+		secretNamespacedName = types.NamespacedName{Name: commonSpec.GetServerSSLCertSecretName(), Namespace: clusterNamespace}
 	}
-	serverName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerServerCertTemplate, r.KafkaCluster.Name), Namespace: r.KafkaCluster.Namespace}
 	serverSecret := &corev1.Secret{}
-	if err := r.Client.Get(context.TODO(), serverName, serverSecret); err != nil {
-		if apierrors.IsNotFound(err) {
-			return "", "", nil, errorfactory.New(errorfactory.ResourceNotReady{}, err, "server secret not ready")
+	if err := client.Get(context.TODO(), secretNamespacedName, serverSecret); err != nil {
+		if apierrors.IsNotFound(err) && commonSpec.GetServerSSLCertSecretName() == "" {
+			return nil, errorfactory.New(errorfactory.ResourceNotReady{}, err, "server secret not ready")
 		}
-		return "", "", nil, errors.WrapIfWithDetails(err, "failed to get server secret")
+		return nil, errors.WrapIfWithDetails(err, "failed to get server secret")
 	}
-	serverPass := string(serverSecret.Data[v1alpha1.PasswordKey])
-
-	clientName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerControllerTemplate, r.KafkaCluster.Name), Namespace: r.KafkaCluster.Namespace}
-	clientSecret := &corev1.Secret{}
-	if err := r.Client.Get(context.TODO(), clientName, clientSecret); err != nil {
-		if apierrors.IsNotFound(err) {
-			return "", "", nil, errorfactory.New(errorfactory.ResourceNotReady{}, err, "client secret not ready")
+	// Check secret data fields
+	if !IsGeneratedSSLCertSecretFilled(*serverSecret) {
+		if commonSpec.GetServerSSLCertSecretName() != "" {
+			return nil, errors.Errorf("secret: %s has missing data fields", serverSecret.Name)
 		}
-		return "", "", nil, errors.WrapIfWithDetails(err, "failed to get client secret")
+		return nil, errorfactory.New(errorfactory.ResourceNotReady{}, errors.Errorf("SSL JKS certificate has not generated properly yet into secret: %s", serverSecret.Name), "checking secret data fields")
 	}
-	clientPass := string(clientSecret.Data[v1alpha1.PasswordKey])
 
-	superUsers := make([]string, 0)
-	for _, secret := range []*corev1.Secret{serverSecret, clientSecret} {
-		cert, err := certutil.DecodeCertificate(secret.Data[corev1.TLSCertKey])
-		if err != nil {
-			return "", "", nil, errors.WrapIfWithDetails(err, "failed to decode certificate")
+	return serverSecret, nil
+}
+
+func (r *Reconciler) getServerPasswordKeysAndUsers() (map[string]string, []string, error) {
```

Comment:
The implementation of this function seems a bit complex. Could you describe it and add some note here what this method actually does?